File: src/main/java/priv/bajdcc/LL1/syntax/exp/IExpCollction.java
package priv.bajdcc.LL1.syntax.exp;

import priv.bajdcc.LL1.syntax.ISyntaxComponent;

/**
 * 可以添加孩子结点的表达式接口
 *
 * @author bajdcc
 */
public interface IExpCollction {
	/**
	 * 添加孩子结点
	 *
	 * @param exp 子表达式
	 */
	void add(ISyntaxComponent exp);

	/**
	 * 集合是否为空
	 *
	 * @return 集合是否为空
	 */
	boolean isEmpty();
}

File: src/main/java/com/bajdcc/LL1/syntax/exp/IExpCollction.kt
package com.bajdcc.LL1.syntax.exp

import com.bajdcc.LL1.syntax.ISyntaxComponent

/**
 * 可以添加孩子结点的表达式接口
 *
 * @author bajdcc
 */
interface IExpCollction {

    /**
     * 集合是否为空
     *
     * @return 集合是否为空
     */
    val isEmpty: Boolean

    /**
     * 添加孩子结点
     *
     * @param exp 子表达式
     */
    fun add(exp: ISyntaxComponent)
}
